### DISCUSSION TITLE
MathJax default URL

### DISCUSSION THREAD

**Discussion by twsh** [jgm/pandoc#9892] [Q&A] [answered]
I am using Pandoc 3.1.3 (because it's what Ubuntu gives me) with the following command: `pandoc --standalone --mathjax --output=foo.html foo.md`. In my HTML I see this:

```
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script src="/usr/share/javascript/mathjax/MathJax.js" type="text/javascript"></script>
```

This doesn't display my maths, presumably because I don't have a local copy of MathJax.

I would have expected this:

```
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>
```

To get that, I have to use `--mathjax=https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js`.

Is this something that changed in a later version of Pandoc? Or am I wrong to expect the link to the CDN when I don't specify it?

**Comment by jgm** [comment, marked as answer]
Stock pandoc will give you an https link for mathjax.
My guess is that Debian/Ubuntu modified this to use their locally installed version.  There's probably a package you can install that will give this to you.

**Reply by jgm** [reply]
You can always install the deb we provide.

**Reply by twsh** [reply]
Thank you! After you suggested that I checked and saw that there is something in the package description about installing the package [libjs-mathjax](https://packages.ubuntu.com/noble/libjs-mathjax).

**Reply by jgm** [reply]
In my opinion, it's a weird decision for Debian.  After all, you're probably going to be using pandoc to create an HTML file to serve on a website, not locally, and the `/usr/.../` URL seems inappropriate for that use.

**Reply by twsh** [reply]
That's how I generally use it, anyway.

**Reply by twsh** [reply]
I found this in README.Debian, via the changelog for the package when they made this choice in 2018, which explains the packager's reasoning:

> Some output formats depend on external material,
e.g. Javascript code for rendering mathematical equations.
>
> Pandoc commonly links to such external material hosted on a public CDN.
Due to concerns for potential breach of privacy,
Debian-packaged Pandoc instead links to locally packaged material.

https://packages.ubuntu.com/noble/pandoc
